### PR TITLE
Added article command

### LINKED ISSUE #218: Add !article command
Add an `!article` command. When a user uses the `!article` command, Horace should respond with a link to the most recent KOA article.
Specific copy for this command is pending, but can be finalized once someone is assigned to the issue.

### PR DESCRIPTION
**Pull request by mukkuzzz**
Added article Command

### CODE REVIEW

**Comment by debashisbiswas** [comment]
Thanks for your changes! Before these get merged, we're double-checking to clarify the intended behavior of the command. We might need it to link to the most recent article, rather than all of the articles, so I'll let you know when I hear back on this.

**Comment by debashisbiswas** [comment]
Until then, would you be able to fix the indentation/whitespace/spacing/missing newline in `article.test.js`? Should be a quick fix.

**Comment by mukkuzzz** [comment]
> Until then, would you be able to fix the indentation/whitespace/spacing/missing newline in `article.test.js`? Should be a quick fix.

DONE

**Comment by mukkuzzz** [comment]
@debashisbiswas hey u can review this now??

**Comment by debashisbiswas** [comment]
Still waiting to hear back about the indented behavior.

File: .gitignore [hunk 1]
```diff
@@ -116,3 +116,6 @@ public
 
 # Database
 *.sqlite
+
+#test out file
+out.txt
```
Comment (debashisbiswas) [inline .gitignore]: Would you be able to remove this change? This doesn't need to get merged.

File: commands/article.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const config = require('../config.json');
+const Parser = require('rss-parser');
+const parser = new Parser();
+const random = require('random');
```
Comment (debashisbiswas) [inline commands/article.js]: I don't think we need to add another dependency for this, can you implement this with `Math.random()` instead?

File: commands/article.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+
+module.exports.execute = async (client, message, args) => {
+	const feed = await parser.parseURL('https://knightsofacademia.org/category/articles/feed');
+	const r = random.int(0 , feed.items.length);
+	let response = `Here is a article i found called "${(feed.items[r].title)}" here is the link to it ${feed.items[r].link}`;
```
Comment (debashisbiswas) [inline commands/article.js]: It's not necessarily clear to the user that this article was randomly selected. How about something like:
`Here's a random article from KOA titled "${(feed.items[r].title)}": ${feed.items[r].link}`

File: commands/article.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	if(args[0]){
+		if(args[0].toLowerCase()==='home'){
+			response = "This is home page of the articles https://kingsofacademacia.org/category/articles";
```
Comment (debashisbiswas) [inline commands/article.js]: This is not the correct link to the website. Use `https://knightsofacademia.org/category/articles`

Also, grammar - how about something like this for consistency with other commands?
`KOA Article Archive: https://knightsofacademia.org/category/articles`

File: commands/article.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+		}
+		else if(args[0].toLowerCase() === 'latest'){
+			response = `Here is the latest article i found called "${(feed.items[0].title)}" here is the link to it ${feed.items[0].link}`
```
Comment (debashisbiswas) [inline commands/article.js]: Capitalization/grammar, how about this:
`The latest article from KOA is called "${(feed.items[0].title)}" and can be found here: ${feed.items[0].link}.`

File: package.json [hunk 6]
```diff
@@ -22,6 +22,7 @@
     "habitica": "^4.0.0",
     "mysql2": "^2.0.2",
     "quick.db": "^7.1.3",
+    "random": "^3.0.5",
```
Comment (debashisbiswas) [inline package.json]: See above comment about unnecessary dependency

**Review by debashisbiswas — CHANGES_REQUESTED** [review]
Sorry for all the comments! Most of them are requests for changes to some of the strings for grammar and capitalization, and you can read the comments for details on the other requested changes.

**Comment by mukkuzzz** [comment]
OK i will change them

**Comment by mukkuzzz** [comment]
@debashisbiswas Changes done

**Review by debashisbiswas — APPROVED** [review]
Great, thanks for all the changes. Looks great!

**Comment by mukkuzzz** [comment]
> Great, thanks for all the changes. Looks great!

Thanks.

**Pull request merged**